devito/passes/clusters/asynchrony.py

```diff
@@ -48,14 +47,12 @@ def callback(self, clusters, prefix):
 
         d = prefix[-1].dim
 
-        if not all(SEQUENTIAL in c.properties[d] for c in clusters):
```

Comment:
Why can this be dropped?

Reply:
it doesn't really matter whether time is parallel or sequential. The time dimensions is, regardless of its computational nature, designated to be the dimension along which streaming occurs. "Designated" -> I mean by the compiler callbacks, which one could in theory customize, but why on earth :)

So I've made things a little more general...

This is because I had a simple test with `Op(Eq(usave, time))` where `time` clearly is parallel but streaming wouldn't trigger because of that annoying condition